### PR TITLE
Display password fields as a password input

### PR DESCRIPTION
**Pull request by huwshimi**

### CODE REVIEW

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/5973/
Test PASSed.

**Comment by huwshimi** [comment]
QA:
- deploy on a cloud that users user/pass or just change one of the cloud types to "demonstration" in deployment-flow.js
- open the add creds form
- the password field should show dots instead of your password

File: jujugui/static/gui/src/app/components/generic-input/test-generic-input.js [hunk 1]
```diff
@@ -66,6 +66,40 @@ describe('GenericInput', function() {
     assert.deepEqual(output, expected);
   });
 
+  it('can display as a different type', () => {
+    var renderer = jsTestUtils.shallowRender(
```
Comment (frankban) [inline jujugui/static/gui/src/app/components/generic-input/test-generic-input.js]: const, here and on all var instances below.

**Comment by jujugui** [comment]
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/5994/
Test FAILed.

**Comment by jujugui** [comment]
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/5995/
Test FAILed.

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/5996/
Test PASSed.

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/5997/
Test PASSed.

**Comment by hatched** [comment]
:shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci.jujugui.org:8080/job/juju-gui-merge

**Pull request merged**